hack/lint2check/README.md

```diff
@@ -0,0 +1,4 @@
+# lint2check
```

Comment:
So we have a tool that runs in a github action and talks to the github checks api to tell it about its own status? This feels super redundant.

Reply:
I agree. I just follow what was defined here: https://github.com/kubernetes-sigs/controller-runtime/issues/501

However, IHMO would not be required the hack/lint2check dir at all. 

@DirectXMan12 could we remove `.github/main.workflow` and the `hack/lint2check` and just leave the lint be called to check the comments as it has been in the `make test` ?

Reply:
+1 on not needing this, we just need to enable the extra checks in golanci-lint

ref: https://github.com/kubernetes-sigs/controller-runtime/pull/523#discussion_r325396374

Reply:
Hi @vincepri, 

I understand that all that needs to be made is done here. I indeed push a test commit to seeing the CI failing. See: https://github.com/kubernetes-sigs/controller-runtime/pull/885#discussion_r404006333

So, did you see anything that is missing be done here?

Reply:
Current changes look good to me, I think it fixes the immediate issue